backend/src/Database.hs

```diff
@@ -122,7 +122,9 @@ toFilterExpression report spec = foldr ((&&.) . fromMaybe (val True)) (val True)
             &&. ((report ^. GameReportWinnerId ==. val player2) ||. (report ^. GameReportLoserId ==. val player2))
     winnerFilter = toFilter report GameReportWinnerId spec.winners
     loserFilter = toFilter report GameReportLoserId spec.losers
-    leagueFilter = toFilter report GameReportLeague (map Just <$> spec.leagues)
+    leagueFilter = case spec.leagues of
```

Comment:
I worry slightly about the semantics of our FilterSpec type getting too tricky/complicated as we continue to add these kinds of corner cases - but I think this is excellent for now.

Reply:
Sweet! Yeah totally see that, it doesn't feel super extensible. Will be on the lookout for refactor opportunities as we add more filters 💭